src/utils/integration/jira.ts

```diff
@@ -0,0 +1,226 @@
+import assert from 'assert';
+import JiraApi from 'jira-client';
+
+export enum AvatarSize {
+    x1 = 16,
+    x2 = 24,
+    x3 = 32,
+    x4 = 48,
+}
+
+export type Avatar<S extends AvatarSize, K extends string = `${S}x${S}`> = Record<K, string>;
+
+export type JiraAvatar = Avatar<AvatarSize.x1> | Avatar<AvatarSize.x2> | Avatar<AvatarSize.x3> | Avatar<AvatarSize.x4>;
+
+export interface JiraUser {
+    active: boolean;
+    displayName: string;
+    emailAddress: string;
+    key: string;
+    name: string;
+    timeZone: string;
+    self: string;
+    avatarUrls: JiraAvatar;
+}
+
+export interface JiraIssueStatus {
+    self: string;
+    description: string;
+    iconUrl: string;
+    name: string;
+    id: string;
+    statusCategory: {
+        self: string;
+        id: number;
+        key: string;
+        colorName: string;
+        name: string;
+    };
+}
+
+export interface JiraProject {
+    self: string;
+    id: string;
+    key: string;
+    name: string;
+    projectTypeKey: string;
+    avatarUrls: JiraAvatar;
+}
+
+export interface JiraIssueType {
+    self: string;
+    id: string;
+    description: string;
+    iconUrl: string;
+    name: string;
+    subtask: boolean;
+    avatarId: number;
+}
+
+export interface JiraPriority {
+    self: string;
+    iconUrl: string;
+    name: string;
+    id: string;
+}
+
+export interface JiraIssue {
+    id: string;
+    key: string;
+    self: string;
+    summary: string;
+    description: string;
+    creator: JiraUser;
+    assignee: JiraUser;
+    reporter: JiraUser;
+    status: JiraIssueStatus;
+    project: JiraProject;
+    priority: JiraProject;
+    issuetype: JiraIssueType;
+}
+
+const isDebugEnabled = process.env.NODE_ENV === 'development' && process.env.DEBUG?.includes('service:jira');
+
+// @ts-ignore
+class JiraService extends JiraApi {
+    private positiveFinishedStatuses: string;
+
+    private finishedStatusCategory: JiraIssueStatus['statusCategory'];
+
+    constructor(options: JiraApi.JiraApiOptions) {
+        super(options);
+
+        assert(
+            process.env.JIRA_POSITIVE_STATUS_NAMES,
+            `Env variable \`JIRA_POSITIVE_STATUS_NAMES\` must be string, but get ${typeof process.env
+                .JIRA_POSITIVE_STATUS_NAMES}`,
+        );
+        assert(
+            process.env.JIRA_FINISHED_CATEGORY,
+            `Env variable \`JIRA_FINISHED_CATEGORY\` must be string, but get ${typeof process.env
+                .JIRA_FINISHED_CATEGORY}`,
+        );
+
+        this.positiveFinishedStatuses = process.env.JIRA_POSITIVE_STATUS_NAMES;
```

Comment:
why it is not array?

Reply:
'cause it is string `"\['value1', ...\]"` anyway